python/pyspark/sql/context.py

```diff
@@ -50,18 +50,56 @@
 def _monkey_patch_RDD(sqlContext):
     def toDF(self, schema=None, sampleRatio=None):
         """
-        Converts current :class:`RDD` into a :class:`DataFrame`
+        Converts current :class:`RDD` into a :class:`DataFrame` according to the given schema.
+        If the given schema is None or just column names, the schema will be inferred by scanning
+        the data in this RDD according to the sampleRatio.
 
-        This is a shorthand for ``sqlContext.createDataFrame(rdd, schema, sampleRatio)``
+        Note that, the given schema must match the real data, or exception will be thrown at
+        runtime. If the given schema is not StructType, it will be wrapped into a StructType as
+        its only field, and the field name will be "value", each record will also be wrapped into a
+        tuple, which can be converted to row later.
 
-        :param schema: a StructType or list of names of columns
+        :param schema: a :class:`DataType` or a datatype string or list of names of columns
         :param samplingRatio: the sample ratio of rows used for inferring
         :return: a DataFrame
 
+        .. versionchanged:: 2.0
+           The schema parameter can be a DataType or a datatype string after 2.0. If it's not a
+           StructType, it will be wrapped into a StructType and each record will also be wrapped
+           into a tuple.
+           The data type string format equals to `DataType.simpleString`, except that top level
+           struct type can omit the `struct<>` and numeric types use `typeName()` as their format,
+           e.g. use `byte` instead of `tinyint` for ByteType. We can also use `int` as a short name
+           for IntegerType.
+
         >>> rdd.toDF().collect()
         [Row(name=u'Alice', age=1)]
+        >>> rdd.toDF("a: string, b: int").collect()
```

Comment:
Could you add another example to show that the schema is a list of column names?